azurelinuxagent/ga/monitor.py

```diff
@@ -172,14 +197,14 @@ class MonitorHandler(ThreadHandlerInterface): # pylint: disable=R0902
     def get_thread_name():
         return MonitorHandler._THREAD_NAME
 
-    def __init__(self):
+    def __init__(self, enqueue_event_func):
```

Comment:
the MonitorHandle shouldn't take the event function as parameter... the monitor is just a collection of arbitrary periodic operations and the details for each of those operations should be internal to them

Reply:
The main reason for doing this was trying to maintain less state. I do agree to your point but at the same time I figured it would be easier to pass out parameter rather than make the Queue a static variable (which is what this function is primary used for at the end of the day). 

1 way to mitigate this I can think of is to move CollectAndEnqueueEventsPeriodicOperation to the extension_telemetry thread. Basically have a single thread for collecting all kinds of events from disk - Agent and extensions. Does that sound like a reasonable approach?

Reply:
You could instantiate CollectAndEnqueueEventsPeriodicOperation in the main thread and expose a method in the MonitorHandle to add an item to the operations it monitors.

Independently of this, I mentioned before that having a single thread for telemetry seemed a better approach so I like your suggestion :)

Reply:
Sounds good, let me do that. Would definitely break a lot of tests lol

Reply:
+1 for keeping the telemetry stuff in a single thread, would make it more clean and thread-safe!

Reply:
After our last sync, we decided to keep 2 threads - One for collecting telemetry data (extension + agent) and another for sending data (telemetry service)

Reply:
Done